src/schema/v2/CollectorProfile/collectorProfiles.ts

```diff
@@ -31,13 +31,16 @@ export const CollectorProfilesConnection: GraphQLFieldConfig<
     { term, partnerID, ...args },
     { collectorProfilesLoader }
   ) => {
-    if (!collectorProfilesLoader)
+    if (!collectorProfilesLoader) {
       throw new Error(
         "A X-Access-Token header is required to perform this action."
       )
-
-    if (!partnerID || !term) {
-      throw new Error("Arguments `partnerID` and `term` are required.")
+    }
+    if (!partnerID) {
+      throw new Error("Argument `partnerID` is required.")
+    }
+    if (term == undefined) {
```

Comment:
**question:** Why do we send `undefined` to Metaphysics? Can we handle this in the UI instead?

Reply:
we never actually receive an `undefined`, if the arg is not present then JS considers the var value `undefined`, and we might send `null` from the client too.
This check captures if it's missing or if the client sends `null`, take a look at the `==` and not the `===`.

Reply:
I think we are sending an empty string when this request fails
![image](https://github.com/artsy/metaphysics/assets/1176374/8f36bef7-468d-4637-a853-3d34d268d41a)
![Screenshot 2024-05-08 at 09 45 11](https://github.com/artsy/metaphysics/assets/1176374/8f76a34a-4285-4464-bd6e-8cb8f4d41ba8)


Why are we sending this during the initial render? Shouldn't it be handled on the client itself?

Reply:
well, that's debatable 😅 
I agree that the clients *could/should* prevent this kind of issue, but that's how the backend is designed to work when receiving `term: ''` and I don't think we should change it right now.
If you strongly feel that we should only do it in volt-v2 I can change it there, or we could even do the combo, the change there and here; I will leave the choice to you 😬

Reply:
I think we shouldn't send anything from the client before user interaction, what do you think?

Reply:
I can't entirely agree 😂 

Mostly because Relay is mainly declarative, so we can't block the query until the field has the `term` to call it(there are some ways of doing it but it's not something we should be doing for this case)

Reply:
I can't entirely agree either 😅 
I don't think we have the filter drawer presented on the first load. I was thinking if we could prevent that from happening.
Anyway - I'll leave this decision to you!

Reply:
it's rendered in a portal

<img width="1504" alt="image" src="https://github.com/artsy/metaphysics/assets/15792853/fc33efb9-7b49-49f4-9af2-8be9057684e9">

Reply:
I will dig more 30-60 minutes into Relay docs to see if I catch something we could use(their doc sucks btw 😞 ) instead of this change here